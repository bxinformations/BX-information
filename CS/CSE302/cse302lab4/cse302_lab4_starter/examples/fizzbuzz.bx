// fizz buzz
// print numbers from 0 to 100,
//   but replace every multiple of 3 with 333
//           and every multiple of 5 with 555
//   except replace multiples of both 3 and 5 with 151515

def fizzbuzz(cur, last : int) {
  while (cur <= last) {
    if (cur % 3 == 0) {
      if (cur % 5 == 0) {
         print(151515);
      } else {
         print(333);
      }
    } else if (cur % 5 == 0) {
      print(555);
    } else {
      print(cur);
    }
    cur = cur + 1;
  }
}

def main() {
  fizzbuzz(0, 100);
}