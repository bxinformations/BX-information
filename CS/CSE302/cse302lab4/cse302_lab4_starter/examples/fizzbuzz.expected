151515
1
2
333
4
555
333
7
8
333
555
11
333
13
14
151515
16
17
333
19
555
333
22
23
333
555
26
333
28
29
151515
31
32
333
34
555
333
37
38
333
555
41
333
43
44
151515
46
47
333
49
555
333
52
53
333
555
56
333
58
59
151515
61
62
333
64
555
333
67
68
333
555
71
333
73
74
151515
76
77
333
79
555
333
82
83
333
555
86
333
88
89
151515
91
92
333
94
555
333
97
98
333
555
